website/notebookconvert.py

```diff
@@ -1,39 +1,57 @@
 import os
 import re
 
+
 def add_header_to_markdown(folder, md):
     name = md[:-3]
-    with open(os.path.join(folder, md), 'r+', encoding='utf-8') as f:
+    with open(os.path.join(folder, md), "r+", encoding="utf-8") as f:
         content = f.read()
         f.truncate(0)
-        content = re.sub(r'style=\"[\S ]*?\"', '', content)
-        content = re.sub(r'<style[\S \n.]*?</style>', '', content)
+        content = re.sub(r"style=\"[\S ]*?\"", "", content)
+        content = re.sub(r"<style[\S \n.]*?</style>", "", content)
         f.seek(0, 0)
         f.write("---\ntitle: {}\nhide_title: true\nstatus: stable\n---\n".format(name) + content)
         f.close()
 
-def convert_notebook_to_markdown(file_path, outputdir):
-    print("Converting {} into markdown \n".format(file_path))
-    convert_cmd = 'jupyter nbconvert --output-dir=\"{}\" --to markdown \"{}\"'.format(outputdir, file_path)
+
+def convert_notebook_to_markdown(folder, nb, outputdir):
+    file_path = os.path.join(folder, nb)
+    print(f"Converting {file_path} into markdown")
+
+    # If the notebook contains cell outputs such as figures, a folder containing cell output images is generated alongside the markdown file
+    # By default, both the folder and files contain the notebook name. But spaces in the notebook name create linking errors in the generated markdown
+    # Therefore, we first generate the markdown file, output folder, and output files with no spaces
+    nb_no_spaces = nb.replace(" ", "").replace(".ipynb", "")
+
+    convert_cmd = f'jupyter nbconvert --output-dir="{outputdir}" --NbConvertApp.output_base="{nb_no_spaces}" --to markdown "{file_path}"'
     os.system(convert_cmd)
 
+    # Afterwards, we rename the generated markdown file to ensure that the markdown file has the same name as notebook
+    md_no_spaces = os.path.join(outputdir, f"{nb_no_spaces}.md")
+    md_final = os.path.join(outputdir, nb.replace(".ipynb", ".md"))
+    print(f"Renaming {md_no_spaces} to {md_final}")
+    os.rename(md_no_spaces, md_final)
+    print()
+
+
 def convert_allnotebooks_in_folder(folder, outputdir):
 
     dic = {
         "CognitiveServices - Overview": os.path.join(outputdir, "features"),
         "Classification": os.path.join(outputdir, "examples", "classification"),
         "CognitiveServices": os.path.join(outputdir, "examples", "cognitive_services"),
+        "DataBalanceAnalysis": os.path.join(outputdir, "examples", "responsible_ai"),
         "DeepLearning": os.path.join(outputdir, "examples", "deep_learning"),
-        "Interpretability": os.path.join(outputdir, "examples", "model_interpretability"),
+        "Interpretability": os.path.join(outputdir, "examples", "responsible_ai"),
+        "ModelInterpretability": os.path.join(outputdir, "examples", "responsible_ai"),
```

Comment:
Should "ModelInterpretability" be put into "features"?

Reply:
In an above comment in this PR, @memoryz requested that we move the Mode Interpretability notebooks (more specifically "ModelInterpretability - Snow Leopard Detection.md" to `examples` instead of `features` because it is more like an example :smile: